WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -56,6 +56,15 @@
     @Inject PostStore mPostStore;
     @Inject SiteStore mSiteStore;
 
+    private class UploadingPost {
+        PostModel postModel;
+        List<MediaModel> pendingMedia = new ArrayList<>();
+        UploadingPost(PostModel postModel, List<MediaModel> pendingMedia) {
+            this.postModel = postModel;
+            this.pendingMedia = pendingMedia;
+        }
+    }
+
```

Comment:
I think we could dispose of the new structure `UploadingPost`, as all we need is the `postId` and the related `pendingMedia` to it. So we could replace `List<UploadingPost> sPostsWithPendingMedia`  with `ConcurrentHashMap<Integer, List<MediaModel>>` as in `sCompletedMediaByPost`.